File: twoSum2/twoSum2.py
def twoSum2(numbers, target):
    l, r = 0, len(numbers) - 1

    while l < r:
        curr = numbers[l] + numbers[r]
        if curr > target:
            r -= 1
        elif curr < target:
            l += 1
        else:
            return [l+1, r+1]

print(twoSum2([1, 3, 4, 5, 7, 11], 9))

File: twoSum2/twoSum2.js
twoSum2 = (numbers, target) => {
    let l = 0
    let r = numbers.length - 1


    while(l < r) {
        let curr = numbers[l] + numbers[r]
        console.log('Left:', l, 'Right:', r)
        console.log(curr)
        if(curr > target) {
            r--
        } else if(curr < target) {
            l++
        } else {
            return [l+1, r+1]
        }
    }
}

console.log(twoSum2([1, 3, 4, 5, 7, 11], 9))
